source/GameInterface/Services/Workshops/WorkshopRegistry.cs

```diff
@@ -0,0 +1,21 @@
+﻿using GameInterface.Services.Registry;
+using System;
+using TaleWorlds.CampaignSystem.Settlements.Workshops;
+
+namespace GameInterface.Services.Workshops
+{
+    internal class WorkshopRegistry : RegistryBase<Workshop>
+    {
+        public WorkshopRegistry(IRegistryCollection collection) : base(collection) { }
+
+        public override void RegisterAll()
+        {
+            //Not needed
```

Comment:
A register all here will be needed as there are Workshops registered on start.

It would look like this. Let me know if you need any more information or have any questions!

`        public override void RegisterAll()
        {
            foreach(Town town in Town.AllTowns)
            {
                foreach(Workshop workshop in town.Workshops)
                {
                    RegisterNewObject(workshop, out var _);
                }
            }
        }`